bin/binaries/micro-python

```diff
@@ -16,6 +16,9 @@ def __prepare():
 __prepare()
 del __prepare
 
+# inject microcule WSGI handler code
+code = code + '\nimport microcule\nmicrocule.wsgi(Hook).run(app)\n'
```

Comment:
This will break all "old-style" python hooks.

Reply:
Good catch, that is correct. I was consider the best ways to fix this.

I think we may have to create a separate code path / type for WSGI enabled applications. This way, users could spawn python functions in either the original way ( write directly to STDOUT ) or the new way of using WSGI style interfaces. We would only run this code injection line if we knew the user-code was defining WSGI style `app` interface.

I would be open to entirely removing the old python way, but I think we may have some backwards compatibility issues, so it would be worth supporting both.

Reply:
Why not leave as is? Tail construction `if __name__…`is standard in python world.

Reply:
The issue I had was that by leaving it, all user-defined functions now much import and call `microcule` python module. That breaks compatibility with some of the python WSGI servers I've seen.

Wouldn't it be better to have user-defined python functions work across multiple WSGI providers?

Reply:
This is limitation of single-file concept. In most cases WSGI-server is adapter between server-specific API (CGI, Apache module system, lighttpd plugin system) and WSGI compatible app. Python module `microcule` falls into same category. In other cases we can split WSGI-application object and WSGI wrapper to different files (sometimes it's just config), but not in our case.
We either should separate types of old-style and wsgi python hooks or provide method of wrapping as a module.